lib/grp.gi

```diff
@@ -1586,6 +1586,35 @@ function( G )
     return comp;
 end );
 
+#############################################################################
+##
+#M  Socle( <G> )  . . . . . . . . . . . . . . . . for finite nilpotent groups
+##
+InstallMethod(Socle, "for nilpotent groups",
+              [ IsGroup and IsFinite and IsNilpotentGroup ],
+  function(G)
+    local p, H, prodH;
+
+    prodH := TrivialSubgroup(G);
+    # now socle is product of Omega of the center of its Sylow subgroups
+    for H in SylowSystem(G) do
+      p := PrimePGroup(H);
+      prodH := ClosureSubgroupNC(prodH, Omega(Center(H), p));
+
+      # prodH is central in G, set some properties and attributes accordingly
+      SetIsAbelian(prodH, true);
+      if not HasParent(prodH) then
+        SetParent(prodH, G);
+        SetCentralizerInParent(prodH, G);
+        SetIsNormalInParent(prodH, true);
+      elif IsSubgroup(G, Parent(prodH)) then
+        SetCentralizerInParent(prodH, Parent(prodH));
+        SetIsNormalInParent(prodH, true);
+      fi;
+    od;
```

Comment:
You created prodH as a subgroup of G here, so I am pretty sure (but have not verified) that G is always the parent of prodH here -- so you could simplify the code a lot.

Moreover, I don't think it's worth it setting this information for the intermediate groups prodH which you are throwing away in the next loop operation anyway. It should be enough to set this information for the final value of `prodH` only, i.e. outside the loop. Or am I missing something?